File: src/neetcode/Java/_26RemoveDuplicatesFromSortedArray.java
package neetcode.Java;

public class _26RemoveDuplicatesFromSortedArray {
    public int removeDuplicates(int[] nums) {
        int k = 1;
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] > nums[i - 1]) {
                nums[k] = nums[i];
            } else {
                k++;
            }
        }
        return k;
    }

    public int removeDuplicates1(int[] nums) {
        int k = 1;
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] == nums[i - 1]) {

            } else {
                nums[k++] = nums[i];
            }
        }
        return k;
    }

    public static void main(String[] args) {
        _26RemoveDuplicatesFromSortedArray obj = new _26RemoveDuplicatesFromSortedArray();
        int[] num = {1, 1, 3};
        System.out.println(obj.removeDuplicates(num));
    }
}

File: src/neetcode/kotlin/_26RemoveDuplicatesFromSortedArray.kt
package neetcode.kotlin

class _26RemoveDuplicatesFromSortedArray {
    fun removeDuplicates(nums: IntArray): Int {
        var k = 1
        for (i in 1 until nums.size) {
            if (nums[i] > nums[i - 1]) {
                nums[k] = nums[i]
            } else {
                k++
            }
        }
        return k
    }

    fun removeDuplicates1(nums: IntArray): Int {
        var k = 1
        for (i in 1 until nums.size) {
            if (nums[i] == nums[i - 1]) {
            } else {
                nums[k++] = nums[i]
            }
        }
        return k
    }

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            val obj = _26RemoveDuplicatesFromSortedArray()
            val num = intArrayOf(1, 1, 3)
            println(obj.removeDuplicates(num))
        }
    }
}
